Duplicate typedef for "bplib_cache_module_api_t"

When compiling with "pedantic" flags (as used by CFS) it triggers an error for redefinition of the typedef:
```
In file included from /home/joe/code/cfecfs/dtncfs/apps/bplib/cache/src/v7_cache_internal.h:28,
                 from /home/joe/code/cfecfs/dtncfs/apps/bplib/cache/src/v7_cache.c:25:
/home/joe/code/cfecfs/dtncfs/apps/bplib/cache/inc/v7_cache.h:64:3: error: redefinition of typedef ‘bplib_cache_module_api_t’ [-Werror=pedantic]
   64 | } bplib_cache_module_api_t;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/joe/code/cfecfs/dtncfs/apps/bplib/inc/bplib.h:32,
                 from /home/joe/code/cfecfs/dtncfs/apps/bplib/cache/src/v7_cache_internal.h:24,
                 from /home/joe/code/cfecfs/dtncfs/apps/bplib/cache/src/v7_cache.c:25:
/home/joe/code/cfecfs/dtncfs/apps/bplib/inc/bplib_api_types.h:86:39: note: previous declaration of ‘bplib_cache_module_api_t’ with type ‘bplib_cache_module_api_t’ {aka ‘struct bplib_cache_module_api’}
   86 | typedef struct bplib_cache_module_api bplib_cache_module_api_t;
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/cache/inc/v7_cache.h
+++ b/cache/inc/v7_cache.h
@@ -53,7 +53,7 @@ typedef enum bplib_cache_confkey
     bplib_cache_confkey_offload_base_dir,
 } bplib_cache_confkey_t;
 
-typedef struct bplib_cache_module_api
+struct bplib_cache_module_api
 {
     bplib_cache_module_type_t module_type;
     bplib_mpool_block_t *(*instantiate)(bplib_mpool_ref_t parent_ref, void *init_arg);
@@ -61,7 +61,7 @@ typedef struct bplib_cache_module_api
     int (*query)(bplib_mpool_block_t *svc, int key, bplib_cache_module_valtype_t vt, const void **val);
     int (*start)(bplib_mpool_block_t *svc);
     int (*stop)(bplib_mpool_block_t *svc);
-} bplib_cache_module_api_t;
+};
 
 typedef struct bplib_cache_offload_api
 {

